plugins/instanceof-checkcast/src/main/java/org/qbicc/plugin/instanceofcheckcast/SupersDisplayBuilder.java

```diff
@@ -33,17 +34,9 @@ public void accept(CompilationContext ctxt) {
         info.visitReachableSubclassesPreOrder(jlo, tables::buildSupersDisplay);
         // Assign typeIDs to classes
         // [0] Poisioned entry for easier debugging
-        // primitives
-        // [1] boolean.class
-        // [2] byte.class
-        // [3] short.class
-        // [4] char.class
-        // [5] int.class
-        // [6] float.class
-        // [7] long.class
-        // [8] double.class
-        // [9] void.class
```

Comment:
The class for primitives does not match this ordering, so it is very important to ensure that we don't have these values hard-coded anywhere. I think every such place would be marked with `TODO` or `XXX` though...

Reply:
Sorry, I didn't get it. Can you please point out where is the mismatch in the ordering?

Reply:
I think the the primitives do match but it's much more implicit now.  The primitives are added with `assignTypeIdToPrimitives()`, which does `PrimitiveType.forEach()` which in turn iterates the `PrimitiveType.TYPES` array using `for (PrimitiveType type : PrimitiveType.TYPES) {`.

As long as the order of elements in the `PrimitiveType.TYPES` doesn't change, this should match.  But it's super subtle.

Best to check everywhere using this and ensure it doesn't rely on the ordering.  (ie: that sub-int types typeid < int.typeid}

Reply:
Sorry, I thought you had `boolean` last, but maybe I just read it wrong without my glasses on.

Reply:
still not clear - why are we worried about order of primitives anyway? is there relationship between typeid of primitives?

Are there any other places where we hard-code the typeid for primitives? Is that what we are are worried about - that the hard-coded values may not match with the order in `PrimitiveType`?

Reply:
Yes, exactly; there may be places where type IDs for primitives were hard-coded so we should ensure that those all are replaced with references to `Primitive`.

Reply:
ok, only place I could find was in `LLVMModuleNodeVisitor` where we lower the `TypeLiteral`. Do you know any other place that needs to be updated.

Reply:
I don't see any other places - the ones that typically need to be changed are the ObjectModel helpers but so far they only care about where Object starts, where the interfaces start, and the arrays.

Reply:
@DanHeidinga thanks for confirming